rpcserver_test.go

```diff
@@ -109,25 +110,44 @@ func TestMain(m *testing.M) {
 	// Initialize the primary mining node with a chain of length 125,
 	// providing 25 mature coinbases to allow spending from for testing
 	// purposes.
-	if err = primaryHarness.SetUp(true, 25); err != nil {
+	if err := primaryHarness.SetUp(true, 25); err != nil {
 		fmt.Println("unable to setup test chain: ", err)
 		os.Exit(1)
 	}
 
 	exitCode := m.Run()
 
-	// Clean up the primary harness created above. This includes removing
-	// all temporary directories, and shutting down any created processes.
-	if err := primaryHarness.TearDown(); err != nil {
-		fmt.Println("unable to setup test chain: ", err)
-		os.Exit(1)
+	// Clean up any active harnesses that are still currently running.This
+	// includes removing all temporary directories, and shutting down any
+	// created processes.
+	if len(rpctest.ActiveHarnesses()) > 0 {
+		if err := rpctest.TearDownAll(); err != nil {
+			fmt.Println("unable to tear down all harnesses: ", err)
+			os.Exit(1)
+		}
 	}
 
 	os.Exit(exitCode)
 }
 
 func TestRpcServer(t *testing.T) {
+	var currentTestNum int
+	defer func() {
+		// If one of the integration tests caused a panic within the main
+		// goroutine, then tear down all the harnesses in order to avoid
+		// any leaked btcd processes.
+		if r := recover(); r != nil {
+			fmt.Println("recovering from test panic: ", r)
```

Comment:
Using `t.Log` would only print the statements if the `-v` is enabled when running `go test`. This first statement prints the nature of the panic itself which can be very valuable when debugging the test.